src/genesis.rs

```diff
@@ -1,23 +1,123 @@
 use {
-    crate::ValidatorType,
+    crate::{new_spinner_progress_bar, ValidatorType, SUN},
     log::*,
     rand::Rng,
     solana_core::gen_keys::GenKeys,
-    solana_sdk::signature::{write_keypair_file, Keypair},
+    solana_sdk::{
+        native_token::sol_to_lamports,
+        signature::{write_keypair_file, Keypair},
+    },
     std::{
         error::Error,
+        fs::File,
+        io::Read,
         path::{Path, PathBuf},
+        process::Command,
         result::Result,
     },
 };
 
+pub const DEFAULT_FAUCET_LAMPORTS: u64 = 500000000000000000; // from agave/
+pub const DEFAULT_MAX_GENESIS_ARCHIVE_UNPACKED_SIZE: u64 = 1073741824; // from agave/
+pub const DEFAULT_INTERNAL_NODE_STAKE_SOL: f64 = 10.0;
+pub const DEFAULT_INTERNAL_NODE_SOL: f64 = 100.0;
+pub const DEFAULT_BOOTSTRAP_NODE_STAKE_SOL: f64 = 10.0;
+pub const DEFAULT_BOOTSTRAP_NODE_SOL: f64 = 100.0;
+
+fn fetch_spl(fetch_spl_file: &PathBuf) -> Result<(), Box<dyn Error>> {
+    let output = Command::new("bash")
+        .arg(fetch_spl_file)
+        .output() // Capture the output of the script
+        .expect("Failed to run fetch-spl.sh script");
+
+    // Check if the script execution was successful
+    if output.status.success() {
+        Ok(())
+    } else {
+        Err(format!(
+            "Failed to fun fetch-spl.sh script: {}",
+            String::from_utf8_lossy(&output.stderr)
+        )
+        .into())
+    }
+}
+
+fn parse_spl_genesis_file(spl_file: &PathBuf) -> Result<Vec<String>, Box<dyn Error>> {
+    // Read entire file into a String
+    let mut file = File::open(spl_file)?;
+    let mut content = String::new();
+    file.read_to_string(&mut content)?;
+
+    // Split by whitespace
+    let mut args = Vec::new();
+    let mut tokens_iter = content.split_whitespace();
+
+    while let Some(token) = tokens_iter.next() {
+        args.push(token.to_string());
+        // Find flag delimiters
+        if token.starts_with("--") {
+            for next_token in tokens_iter.by_ref() {
+                if next_token.starts_with("--") {
+                    args.push(next_token.to_string());
+                } else {
+                    args.push(next_token.to_string());
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
Sorry if I'm being dense, but why is this logic needed? Shouldn't you be able to just take the `Vec<String>` after the whitespace has been removed?

Reply:
you are right. i had a different (ultimately wrong) understanding of how `solana-genesis` took in these commands. And when I figured out it was wrong, I didn't change this function. But even now realize this function just doesn't really do anything that isn't done by just taking the Vec<String>.